File: Golang/HyechatDemo/heybotclient/model/command.go
package model

const (
	TypeSubCommand      int = 1
	TypeSubCommandGroup int = 2
	TypeString          int = 3
	TypeInteger         int = 4
	TypeBoolean         int = 5
	TypeUser            int = 6
	TypeChannel         int = 7
	TypeRole            int = 8
)

type ChannelBaseInfo struct {
	ChannelId   string `json:"channel_id"`
	ChannelName string `json:"channel_name"`
	ChannelType int    `json:"channel_type"`
}

type CommandInfo struct {
	Id      string     `json:"id"`
	Name    string     `json:"name"`
	Type    int        `json:"type"`
	Options []*Options `json:"options,omitempty" bson:"options"`
}

type Options struct {
	Name    string     `form:"name" json:"name" bson:"name" `
	Type    int        `form:"type" json:"type" bson:"type"`
	Value   string     `form:"value" json:"value,omitempty" bson:"value"`
	Choices []*Options `form:"choices" json:"choices,omitempty" bson:"choices"`
}

File: Python/HeyhcatDemo/conf/command.py
from typing import List, Optional
from pydantic import BaseModel

# Constants
TYPE_SUB_COMMAND = 1
TYPE_SUB_COMMAND_GROUP = 2
TYPE_STRING = 3
TYPE_NUMBER= 4
TYPE_BOOLEAN = 5
TYPE_USER = 6
TYPE_CHANNEL = 7
TYPE_ROLE = 8
TYPE_SELECT = 9
TYPE_INTEGER  = 10


class CommandInfo(BaseModel):
    id: str
    name: str
    type: int
    options: Optional[List['Options']]


class Options(BaseModel):
    value: str
    name: str
    choices: Optional[List['Options']] = None
    type: int
